PR: Add profiler docs to index and improve language (#121)
spyder-ide/spyder-docs: doc/profiler.rst

Running the Profiler
====================

> You can browse for a file using the open button or manually enter the path in the :guilabel:`Profiler`'s path box (top left of the pane). You can then run the analysis on the file by pressing :guilabel:`Profile` in the :guilabel:`Profiler` pane.

Reviewer: ```suggestion
You can browse for a file using the open button or manually enter the path in the :guilabel:`Profiler`'s path box (top left of the pane).
You can then run the analysis on the file by pressing :guilabel:`Profile` in the :guilabel:`Profiler` pane.
```

GIF (selecting file)

> You can also run profiling for the file that is currently open in the :doc:`editor` by
> clicking :menuselection:`&Run --> Profile` in the menu bar, or by using a configurable shortcut (:kbd:`F10` by default).

Reviewer: ```suggestion
You can also run profiling for the file that is currently open in the :doc:`editor` by clicking :menuselection:`&Run --> Profile` in the menu bar, or by using a configurable shortcut (:kbd:`F10` by default).
```

Running the Profiler
====================

You can browse for a file using the open button to the right of the `Profiler`'s path box (top left of the pane), which will run profiling over this file automatically. 

.. image:: images/profiler/profiler-execution-browse.gif
   :alt: Spyder Profiler pane, showing browsing a file and running profiler

|

> You can manually enter the path in the pane's path box and then run the analysis on the file by pressing :guilabel:`Profile` in the :guilabel:`Profiler` pane.

Reviewer: ```suggestion
You can manually enter the path in the pane's path box and then run the analysis on the file by pressing :guilabel:`Profile` in the Profiler pane.
```

Thought we discusssed eliminating the kinda redundant `:guilabel:` directive here? But idk

If you'd like to cancel an in-progress run, click the :guilabel:`Stop` button in the top right, and if profiling fails for any reason, the :guilabel:`Output` dialog will be displayed, indicating the error that occurred. 

> By double-clicking an item in the Profiler, you'll be taken to the file and line in the :doc:`editor` where it was called.

Reviewer: ```suggestion
By double-clicking an item in the Profiler, you will be taken to the file and line in the :doc:`editor` where it was called.
```

Contraction

|

By clicking the dropdown or press the :kbd:`Down Arrow` key in the filename field, you can recall paths of previous profiled files.

.. image:: images/profiler/profiler-previous-paths.png
   :alt: Spyder Profiler pane, showing dropdown of previous profiled files

|

> Finally, you can save the data for a given run to disk as a file with the ``.Result`` extension using the :guilabel:`Save data` button. This can be loaded to compare with a previous run of the same file using the :guilabel:`Load data` button.

Reviewer: ```suggestion
Finally, you can save the data for a given run to disk as a file with the ``.Result`` extension using the :guilabel:`Save data` button.
This can be loaded to compare with a previous run of the same file using the :guilabel:`Load data` button.
```

Line break.

Meanwhile, :guilabel:`Total Time` for ``print_wrapper()`` would be 3 ms, but :guilabel:`Local Time` only 1 ms as the rest of that time was spend inside the ``print()`` function it called.

================
Profiler Plugins
================

There are two additional components that you can install to enable different types of profiling in Spyder. 
First, Spyder Line Profiler allows you to benchmark each line of your code individually. 
To learn more, visit the `spyder-line-profiler git repository`_.

.. _spyder-line-profiler git repository: https://github.com/spyder-ide/spyder-line-profiler

Second, Spyder Memory Profiler measures the memory usage of your code. 
For more information, see the `spyder-memory-profiler git repository`_.

.. _spyder-memory-profiler git repository: https://github.com/spyder-ide/spyder-memory-profiler

> 

Reviewer: ```suggestion
```

|

For example, suppose you ran the :guilabel:`Profiler` on a file calling a function ``sleep_wrapper()`` that in turn called the ``sleep()`` function, and the ``sleep_wrapper()`` function took a total of 3.66 ms to run, with 3.64 ms of that spent executing the ``sleep()`` function inside it.
Therefore, if ``sleep()`` called nothing else itself, its :guilabel:`Total Time` and :guilabel:`Local Time` would both be identical, at 3.64 ms.

> Meanwhile, :guilabel:`Total Time` for ``sleep_wrapper()`` would be 3.66 ms, but :guilabel:`Local Time` only 0.02 ms as the rest of that time was spend inside the ``sleep()`` function it called.

Reviewer: ```suggestion
Meanwhile, :guilabel:`Total Time` for ``sleep_wrapper()`` would be 3.66 ms, but :guilabel:`Local Time` only 0.02 ms as the rest was spent inside the ``sleep()`` function it called.
```

My mistake originally

================
Profiler Plugins
================

> There are two additional third-party plugins that you can install to enable other types of profiling in Spyder. 

Reviewer: ```suggestion
There are two additional plugins that you can install to enable other types of profiling in Spyder. 
```

They aren't third party because the Spyder team ourselves maintains them under our organization. Sorry if it was I that mistakenly suggested or okay'd this, can't remember.

> Meanwhile, :guilabel:`Total Time` for ``print_wrapper()`` would be 3 ms, but :guilabel:`Local Time` only 1 ms as the rest of that time was spend inside the ``print()`` function it called.

Reviewer 2: I think the example in this paragraph should be shown in the image displayed above it; or it should be changed to describe what users are seeing in the image.

In my opinion, that would make the interpretation of results far more easier than it is right now.

================
Profiler Plugins
================

> There are two additional components that you can install to enable different types of profiling in Spyder. 

Reviewer 2: ```suggestion
There are two additional third-party plugins that you can install to enable other types of profiling in Spyder. 
```

* I think we should call them `plugins` instead of `components`.
* Improve phrasing a bit.

First, Spyder Line Profiler allows you to benchmark each line of your code individually. 

> To learn more, visit the `spyder-line-profiler git repository`_.

Reviewer 2: ```suggestion
To learn more about it, please visit the `spyder-line-profiler git repository`_.
```

Improve phrasing.

.. _spyder-line-profiler git repository: https://github.com/spyder-ide/spyder-line-profiler

Second, Spyder Memory Profiler measures the memory usage of your code. 

> For more information, see the `spyder-memory-profiler git repository`_.

Reviewer 2: ```suggestion
For more information, please go to the `spyder-memory-profiler git repository`_.
```

Improve phrasing.
